ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/application/PluginBuilder.kt

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+@file:Suppress("DEPRECATION")
+
+package io.ktor.server.application
+
+import io.ktor.server.application.debug.*
+import io.ktor.server.config.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.util.*
+import io.ktor.util.debug.*
+import io.ktor.util.pipeline.*
+import kotlinx.coroutines.*
+import kotlin.random.*
+
+/**
+ * A builder that is available inside a plugin creation block. You can use this builder to hook in various steps of request processing.
+ **/
+@PluginsDslMarker
+public interface PluginBuilder<PluginConfig : Any> {
```

Comment:
can be sealed interface (too prevent inheriting outside ktor)?

Reply:
Thanks, fixed